minject/inject.py

```diff
@@ -103,6 +148,11 @@ def __init__(self, key: "RegistryKey[T_co]") -> None:
     def resolve(self, registry_impl: Resolver) -> T_co:
         return registry_impl.resolve(self._key)
 
+    async def aresolve(self, registry_impl: Resolver) -> T_co:
+        if _is_key_async(self._key):
+            return await registry_impl._aresolve(self._key)
+        return registry_impl.resolve(self._key)
```

Comment:
question: when the key is not async, we still resolve it by calling the blocking function. Not sure if we discussed it somewhere else, but should we
1. raise an error because the function is used in an unexpected way or
2. do something like `await asyncio.to_thread(registry_impl.resolve, self._key)` to make this really async?

*realized we do throw RegistryAPIError for aget, and so this probably for implementation purpose.

Reply:
Let's merge this discussion with https://github.com/duolingo/minject/pull/49#discussion_r1785068482 . I think these are largely getting at the same thing. Let me know if you disagree though!